.github/workflows/docker.yml

```diff
@@ -0,0 +1,56 @@
+name: docker
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'push' && github.run_number || github.event.pull_request.number }}
+  cancel-in-progress: true
+
+jobs:
+  push-docker-image:
+    name: Build Docker image
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup QEMU
+        uses: docker/setup-qemu-action@v3
+        with:
+          platforms: amd64,arm64
+
+      - name: Setup Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Get latest version
+        id: get-latest-version
+        run: echo "::set-output name=version::$(ls -t build-specs | head -n1)"
+
+      - name: Run build.sh script
+        run: bash build.sh ${{ steps.get-latest-version.outputs.version }}
+
+      - name: Login to Artifactory
+        if: ${{ github.event_name == 'push' }}
+        uses: docker/login-action@v3
+        with:
+          registry: splitio-docker.jfrog.io
+          username: ${{ secrets.ARTIFACTORY_DOCKER_USER }}
+          password: ${{ secrets.ARTIFACTORY_DOCKER_PASS }}
+
+      - name: Docker build
```

Comment:
the build script in the repo is already setup to build and push, so you don't need this step

but docker login should happen before.